core/lib/presentation/views/html_viewer/html_content_viewer_on_web_widget.dart

```diff
@@ -0,0 +1,232 @@
+  String _getRandString(int len) {
+    var random = Random.secure();
+    var values = List<int>.generate(len, (i) => random.nextInt(255));
+    return base64UrlEncode(values);
+  }
+
+  String _generateHtmlDocument(String content) {
+    final htmlScripts = '''
+      <script type="text/javascript">
+        window.parent.addEventListener('message', handleMessage, false);
+      
+        function handleMessage(e) {
+          if (e && e.data && e.data.includes("toIframe:")) {
+            var data = JSON.parse(e.data);
+            if (data["view"].includes("$createdViewId")) {
+              if (data["type"].includes("getHeight")) {
+                var height = document.body.scrollHeight;
+                window.parent.postMessage(JSON.stringify({"view": "$createdViewId", "type": "toDart: htmlHeight", "height": height}), "*");
+              }
+              if (data["type"].includes("execCommand")) {
+                if (data["argument"] === null) {
+                  document.execCommand(data["command"], false);
+                } else {
+                  document.execCommand(data["command"], false, data["argument"]);
+                }
+              }
+            }
+          }
+        }
+      </script>
+    ''';
+
+    final htmlTemplate = '''
+      <!DOCTYPE html>
+      <html>
+      <head>
+      <meta name="viewport" content="width=device-width, initial-scale=1.0">
+      <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
+      <style>
+        #editor {
+          outline: 0px solid transparent;
+          min-height: ${minHeight}px;
+          min-width: 300px;
+          color: #000000;
+          font-family: verdana;
+        }
+        table {
+          width: 100%;
+          max-width: 100%;
+        }
+        td {
+          padding: 13px;
+          margin: 0px;
+        }
+        th {
+          padding: 13px;
+          margin: 0px;
+        }
+      </style>
+      $htmlScripts
+      </head>
+      <body>
+      <div id="editor">$content</div>
+      </body>
+      </html> 
+    ''';
+
+    return htmlTemplate;
+  }
+
+  void _setUpWeb() {
+    _htmlData = _generateHtmlDocument(widget.contentHtml);
+
+    final iframe = html.IFrameElement()
+      ..width = widget.widthContent.toString()
+      ..height = actualHeight.toString()
+      ..srcdoc = _htmlData ?? ''
+      ..style.border = 'none'
+      ..style.overflowY = 'hidden'
+      ..onLoad.listen((event) async {
+        var data = <String, Object>{'type': 'toIframe: getHeight'};
+        data['view'] = createdViewId;
+        final jsonEncoder = JsonEncoder();
+        var jsonStr = jsonEncoder.convert(data);
+        html.window.postMessage(jsonStr, '*');
+
+        html.window.onBeforeUnload.listen((event) {
+          event.preventDefault();
+        });
+
+        html.window.onMessage.listen((event) {
+          var data = json.decode(event.data);
+          developer.log('onMessage(): data: $data', name: 'HtmlContentViewerOnWeb');
```

Comment:
please use `log` from app_logger, because we should dump the error also to the console of browser